File: Cracking the Coding Interview/Data Structures/Hash Tables- Ransom Note/HashTablesRansomNote.java
// https://www.hackerrank.com/challenges/ctci-ransom-note/problem

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int m = in.nextInt();
        int n = in.nextInt();
        Map<String, Integer> magazineCounter = new HashMap<>();
        for(int i=0; i < m; i++){
            String word = in.next();
            magazineCounter.put(word, magazineCounter.containsKey(word) ? magazineCounter.get(word) + 1 : 1);
        }
        Map<String, Integer> ransomCounter = new HashMap<>();
        for(int i=0; i < n; i++){
            String word = in.next();
            ransomCounter.put(word, ransomCounter.containsKey(word) ? ransomCounter.get(word) + 1 : 1);
        }
        boolean yes = true;
        for (String word: ransomCounter.keySet()) {
            if (magazineCounter.get(word) < ransomCounter.get(word)) {
                yes = false;
                break;
            }
        }
        System.out.println(yes ? "Yes" : "No");
    }
}

File: Cracking the Coding Interview/Data Structures/Hash Tables- Ransom Note/HashTablesRansomNote.py
# https://www.hackerrank.com/challenges/ctci-ransom-note/problem

from collections import Counter

def ransom_note(magazine, ransom):
    magazineCounter = Counter(magazine)
    ransomCounter = Counter(ransom)
    numberOfInconsistencies = sum(1 for word, number in ransomCounter.items() if magazineCounter[word] < number)
    return 'No' if numberOfInconsistencies > 0 else 'Yes'

m, n = map(int, input().strip().split(' '))
magazine = input().strip().split(' ')
ransom = input().strip().split(' ')
print(ransom_note(magazine, ransom))
